### PR TITLE
Au test spawning a blocking thread in a transaction hook continuation.

### PR DESCRIPTION
**Pull request by ywkaras**
Getting a result from it, without blocking event thread.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -35,7 +35,7 @@ Copyright (C) 2013 GoDaddy Operating Company, LLC
 
 ~~~
 
-lib/cppapi developed by LinkedIn
+include/tscpp/api developed by LinkedIn
```
Comment (ywkaras) [inline NOTICE:38]: These changes are unrelated cleanup, from moving lib/cppapi to include/tscpp/api and src/tscpp/api .

**Comment by maskit** [comment]
API review on dev@ ?

**Comment by ywkaras** [comment]
> API review on dev@ ?

There is not change to the TS API.

File: NOTICE [hunk 2]
```diff
@@ -35,7 +35,7 @@ Copyright (C) 2013 GoDaddy Operating Company, LLC
 
 ~~~
 
-lib/cppapi developed by LinkedIn
+include/tscpp/api, src/tscpp/api developed by LinkedIn
```
Comment (ywkaras) [inline NOTICE:38]: This included some unrelated cleanup of kruft leftover from lib/cppapi moving to include|src/tscpp/api .

**Comment by maskit** [comment]
>> API review on dev@ ?

> There is not change to the TS API.

plugins/xdebug/Cleanup.h → include/tscpp/api/Cleanup.h

You are introducing a new cpp api. See discussion on #6197.

**Comment by ywkaras** [comment]
> > > API review on dev@ ?
> 
> > There is not change to the TS API.
> 
> plugins/xdebug/Cleanup.h → include/tscpp/api/Cleanup.h
> 
> You are introducing a new cpp api. See discussion on #6197.

OK done.

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by maskit** [comment]
@ywkaras I meant you should send API change proposal on dev@ and I thought you did it, but I couldn't find the mail. What was done? You mean you saw the discussion?

**Comment by ywkaras** [comment]
I tried to, maybe my e-mail is blocked.

**Review by maskit — CHANGES_REQUESTED** [review]
Thanks for proposing the API change on the list. The change should be made separately even if it's accepted.

**Comment by ywkaras** [comment]
> Thanks for proposing the API change on the list. The change should be made separately even if it's accepted.

How would I do that?  I an using definitions in cleanup.h in the test's plugin.

**Comment by ywkaras** [comment]
[approve ci fedora]

**Comment by ywkaras** [comment]
OK @maskit the change you requested has been made.

**Comment by maskit** [comment]
I'll leave actual code review to @rob05c,

**Comment by ywkaras** [comment]
@rob05c should I find someone else to review this?

**Comment by ywkaras** [comment]
@rob05c will you have time to review this?  Should we get someone else?

**Review by maskit — APPROVED** [review]
